### PR TITLE
Avoid potential NoSuchElementException BigQuery filter builder

### PR DESCRIPTION
**Pull request by regadas**
## Description

WHile going through the code notice a potential issue if for some reason the `getDomains()` returns empty `Optional`

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by ebyhr — COMMENTED** [review]
Please add a test to `BaseBigQueryConnectorTest`.

**Comment by regadas** [comment]
@ebyhr will do. I wondered where to add it 😄

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java [hunk 1]
```diff
@@ -145,10 +148,7 @@ else if (singleValues.size() > 1) {
 
     private Optional<String> toPredicate(String columnName, String operator, Object value, BigQueryColumnHandle column)
     {
-        Optional<String> valueAsString = convertToString(column.getTrinoType(), column.getBigqueryType(), value);
-        if (valueAsString.isEmpty()) {
-            return Optional.empty();
-        }
-        return Optional.of(quote(columnName) + " " + operator + " " + valueAsString.get());
+        return convertToString(column.getTrinoType(), column.getBigqueryType(), value)
```
Comment (ebyhr) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java:151]: Revert. This variable is helpful when debugging the code.

Comment (regadas) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java:151]: Yeah, not mixing things up sounds reasonable 👍 I'll update

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java [hunk 2]
```diff
@@ -71,10 +72,12 @@ private List<String> toConjuncts(List<BigQueryColumnHandle> columns)
...
-            Domain domain = tupleDomain.getDomains().get().get(column);
-            if (domain != null) {
-                toPredicate(toBigQueryColumnName(column.getName()), domain, column).ifPresent(clauses::add);
-            }
+            tupleDomain.getDomains()
+                    .filter(domains -> domains.containsKey(column))
+                    .map(domains -> domains.get(column))
+                    .filter(Objects::nonNull)
+                    .flatMap(domain -> toPredicate(toBigQueryColumnName(column.getName()), domain, column))
+                    .ifPresent(clauses::add);
```
Comment (regadas) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java:80]: Actually, I just realized that this is not needed even though it is more correct. Columns are derived from the `TupleDomains`. Given that we only reach this if domains are not empty (we have columns) and it this point, it's safe to call `.get`.

**Comment by regadas** [comment]
Closing this as I realized it's not needed. @ebyhr Thank you so much for the review and sorry about the noise.

**Pull request closed without merging**